src/sql/src/plan.rs

```diff
@@ -574,7 +582,8 @@ pub struct AlterSecretPlan {
 #[derive(Debug)]
 pub struct AlterSystemSetPlan {
     pub name: String,
-    pub value: SetVariableValue,
+    // If Some, the flattened SET values. If None, reset to default.
+    pub value: Option<String>,
```

Comment:
Do I understand the life-cycle of this value correctly?
1. During parsing, we parse this into a list of values.
2. During planning we flatten the list into a single string (or `None`).
3. During sequencing, as part of the catalog transaction, we persist the flattened value to the stash.
4. During sequencing, as part of the catalog transaction, we un-flatten the value and pass it to the catalog to update the in-memory state.

If that's true then it might be slightly cleaner to switch to something like this:
1. During parsing, we parse this into a list of values.
2. During planning we **_leave the value as a list_**.
3. During sequencing, as part of the catalog transaction, we **_flatten the list and_** persist the flattened value to the stash.
4. During sequencing, as part of the catalog transaction, we pass **_the list to_** the catalog to update the in-memory state.

That way we get the following benefits:
- `AlterSystemSetPlan` more closely resembles `SetVariablePlan`.
- We avoid an unnecessary un-flattening.

Reply:
This description leaves out a few steps, which helps explain why the current API exists. We synchronize the system vars in the catalog from the stash, the bootstrap cli args, and launch darkly (https://github.com/MaterializeInc/materialize/blob/d0f12134ccc5193361c2a156ec57e821db100ca4/src/adapter/src/catalog.rs#L2740). Those APIs operate on `String`. I've tried to put the boundary on which things need to parse and flatten such that the lowest level SystemVar type itself gets a `[String]`, but most of the stuff above it is a `String`. There is perhaps a better way, but this way seems fine for now.